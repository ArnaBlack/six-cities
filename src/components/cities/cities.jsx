import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import CityLink from '../city-link/city-link.jsx';

const Cities = ({currentCity, cities}) => <section className="locations container">
  <ul className="locations__list tabs__list">
    {cities.map((it) => <li
      key={`city-${it}`}
      className="locations__item"
    >
      <CityLink
        city={it}
        isActive={it === currentCity}
      />
    </li>)}
  </ul>
</section>;

Cities.propTypes = {
  currentCity: PropTypes.string.isRequired,
  cities: PropTypes.arrayOf(PropTypes.string).isRequired,
};

const mapStateToProps = (state, props) => ({
  ...props,
  currentCity: state.city,
  cities: [...new Set(state.offers.map((it) => it.city))],
});

export {Cities};
export default connect(mapStateToProps)(Cities);
