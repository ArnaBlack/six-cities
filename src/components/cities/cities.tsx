import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import CityLink from '../city-link/city-link';
import {getCurrentCity} from '../../store/data/selectors';
import {getCities} from '../../store/data/selectors';

const Cities = (props) => {
  const {
    currentCity,
    cities,
    onCityClick,
  } = props;

  return <section className="locations container">
    <ul className="locations__list tabs__list">
      {cities.map((it) => <li
        key={`city-${it.name}`}
        className="locations__item"
      >
        <CityLink
          city={it}
          isActive={it.name === currentCity.name}
          onCityClick={onCityClick}
        />
      </li>)}
    </ul>
  </section>;
};

Cities.propTypes = {
  currentCity: PropTypes.shape({
    name: PropTypes.string.isRequired,
    location: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
      zoom: PropTypes.number.isRequired,
    }).isRequired,
  }).isRequired,
  cities: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    location: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
      zoom: PropTypes.number.isRequired,
    }).isRequired,
  })).isRequired,
  onCityClick: PropTypes.func.isRequired,
};

const mapStateToProps = (state, props) => ({
  ...props,
  currentCity: getCurrentCity(state),
  cities: getCities(state),
});

export {Cities};
export default connect(mapStateToProps)(Cities);
