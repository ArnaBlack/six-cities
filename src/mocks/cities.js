export default [
  {
    name: `Paris`,
    coordinates: [48.85341, 2.3488],
  },
  {
    name: `Cologne`,
    coordinates: [50.941357, 6.958307],
  },
  {
    name: `Brussels`,
    coordinates: [50.8505, 4.3488],
  },
  {
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Hamburg`,
    coordinates: [53.551086, 9.993682],
  },
  {
    name: `Dusseldorf`,
    coordinates: [51.217941, 6.761680],
  },
];
