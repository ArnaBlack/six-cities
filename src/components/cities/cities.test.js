import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Cities} from './cities.tsx';

Enzyme.configure({adapter: new Adapter()});

const mock = {
  currentCity: {
    name: `Amsterdam`,
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13,
    },
  },
  cities: [
    {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    {
      name: `Paris`,
      location: {
        latitude: 51.37454,
        longitude: 3.897976,
        zoom: 10,
      },
    }
  ],
};

it(`Cities renders correctly`, () => {
  const {
    currentCity,
    cities,
  } = mock;
  const onCityClick = jest.fn();
  const tree = shallow(<Cities
    currentCity={currentCity}
    cities={cities}
    onCityClick={onCityClick}
  />);

  expect(tree).toMatchSnapshot();
});

