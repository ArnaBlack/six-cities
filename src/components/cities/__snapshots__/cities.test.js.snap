// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities renders correctly 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
      key="city-0"
    >
      <ConnectFunction
        city="Paris"
        isActive={false}
      />
    </li>
    <li
      className="locations__item"
      key="city-1"
    >
      <ConnectFunction
        city="Cologne"
        isActive={false}
      />
    </li>
    <li
      className="locations__item"
      key="city-2"
    >
      <ConnectFunction
        city="Brussels"
        isActive={false}
      />
    </li>
    <li
      className="locations__item"
      key="city-3"
    >
      <ConnectFunction
        city="Amsterdam"
        isActive={true}
      />
    </li>
    <li
      className="locations__item"
      key="city-4"
    >
      <ConnectFunction
        city="Hamburg"
        isActive={false}
      />
    </li>
    <li
      className="locations__item"
      key="city-5"
    >
      <ConnectFunction
        city="Dusseldorf"
        isActive={false}
      />
    </li>
  </ul>
</section>
`;
